### PR TITLE
Generic/EmptyPHPStatement: add XML documentation

### PR DESCRIPTION
**Pull request by rodrigoprimo**
## Description

This PR adds documentation for the Generic/EmptyPHPStatement sniff.

I added one `<standard>` block per warning generated by this sniff. I considered adding another `<standard>` block to the beginning of the file summarizing what this sniff does. Something like:

```xml
<documentation title="Empty PHP Statement">
    <standard>
    <![CDATA[
    Empty PHP tags and superfluous semi-colons are not allowed.
    ]]>
    </standard>
    (...)
</documentation>
```

It does duplicate the content of the other `<standard>` blocks, but it provides a nice summary at the top of the documentation. Without the summary, there is a chance that someone skimming through the documentation might assume that this sniff is only about empty PHP tags and doesn't cover superfluous semi-colons.

I ended up not including the "summary" `<standard>` most of the other sniffs that contain multiple `<standard>` blocks don't contain a summary at the top. The only exception that I found is [ArrayDeclarationStandard.xml](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/e9ebf5253a3e6dd3cc4b584ce54b1f4c34fcb1f3/src/Standards/Squiz/Docs/Arrays/ArrayDeclarationStandard.xml). Any thoughts on this?

## Suggested changelog entry
Add documentation for the Generic.CodeAnalysis.EmptyPHPStatement sniff

## Related issues/external references

Part of #148

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [x] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

**Comment by jrfnl** [comment]
> One of the GitHub actions failed when trying to download PHPCS phar 

This is a known issue and shouldn't happen anymore (for now) - see #204

File: src/Standards/Generic/Docs/CodeAnalysis/EmptyPHPStatementStandard.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <code_comparison>
+        <code title="Valid: There is at least one statement inside the PHP tag pair.">
+        <![CDATA[
+<?php <em>echo 'Hello World';</em> ?>
+<?= <em>'Hello World';</em> ?>
+        ]]>
+        </code>
+        <code title="Invalid: There is no statement inside the PHP tag pair.">
+        <![CDATA[
+<?php <em>// some comment</em> ?>
```
Comment (jrfnl) [inline src/Standards/Generic/Docs/CodeAnalysis/EmptyPHPStatementStandard.xml:16]: Uh.. oh.. did you actually check the example code with the sniff ? You may want to do so...

**Review by jrfnl — CHANGES_REQUESTED** [review]
Thanks for this PR @rodrigoprimo .

> I considered adding another `<standard>` block to the beginning of the file summarizing what this sniff does.
> ...
> It does duplicate the content of the other `<standard>` blocks, but it provides a nice summary at the top of the documentation.
>...
> Any thoughts on this?

I'm not against adding a summary block at the start when it has value, but I don't see it having value when it just repeats what's already in the other blocks.

So:
❌ `Empty PHP tags and superfluous semi-colons are not allowed.`
✔️ `Empty PHP statements are not allowed` (=> the empty PHP tag sets and superfluous semi-colons are _symptoms_, _empty PHP statements is the actual _problem_ the sniff addresses.)

File: src/Standards/Generic/Docs/CodeAnalysis/EmptyPHPStatementStandard.xml (hunk 1 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Docs/CodeAnalysis/EmptyPHPStatementStandard.xml]: I'm sorry, I'm testing all the example codes before creating the documentation but I don't know what happened here. I just fixed it.

**Comment by rodrigoprimo** [comment]
> I'm not against adding a summary block at the start when it has value, but I don't see it having value when it just repeats what's already in the other blocks.

> So:
> ❌ Empty PHP tags and superfluous semi-colons are not allowed.
> ✔️ Empty PHP statements are not allowed (=> the empty PHP tag sets and superfluous semi-colons are symptoms, empty PHP > statements is the actual problem the sniff addresses.)

Thanks for your input. In the end, I decided not to include a summary description at the top. I'm not super sure about and after trying it now it felt a little bit weird and repetitive since there is no clear separation between summary description and the description for the first `<code_comparison>` block.

This PR is ready for a final review.

**Review by jrfnl — APPROVED** [review]
LGTM! Thanks @rodrigoprimo !

**Pull request merged**